src/chains/solana/deposit.ts

```diff
@@ -0,0 +1,111 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Wallet } from "@coral-xyz/anchor";
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { PublicKey } from "@solana/web3.js";
+import GATEWAY_DEV_IDL from "@zetachain/protocol-contracts-solana/dev/idl/gateway.json";
+import GATEWAY_PROD_IDL from "@zetachain/protocol-contracts-solana/prod/idl/gateway.json";
+import { ethers } from "ethers";
+
+import { RevertOptions } from "../../../types/contracts.types";
+import {
+  createRevertOptions,
+  getAPIbyChainId,
+  getSPLToken,
+  isSOLBalanceSufficient,
+} from "../../../utils/solana.commands.helpers";
+
+type solanaDepositParams = {
+  amount: string;
+  receiver: string;
+  revertOptions: RevertOptions;
+  token?: string;
+};
+
+type solanaOptions = {
+  chainId: string;
+  signer: anchor.web3.Keypair;
+};
+
+export const solanaDeposit = async (
+  params: solanaDepositParams,
+  options: solanaOptions
+) => {
+  // Mainnet and devnet use the same IDL
+  const gatewayIDL =
+    options.chainId === "0901" ? GATEWAY_DEV_IDL : GATEWAY_PROD_IDL;
+
+  const API = getAPIbyChainId(options.chainId);
+
+  const connection = new anchor.web3.Connection(API);
+  const provider = new anchor.AnchorProvider(
+    connection,
+    new Wallet(options.signer)
+  );
+  const gatewayProgram = new anchor.Program(gatewayIDL as anchor.Idl, provider);
```

Comment:
This setup looks almost identical for all solana functions, we could probably extract and reuse.

Reply:
0b7bd26